src/utils/skip_list.hpp

```diff
@@ -1036,17 +1036,27 @@ class SkipList final : detail::SkipListNode_base {
 
       TNode *new_node;
       {
-        TNode *prev_pred = nullptr;
+        TNode *previous_locked = nullptr;
         bool valid = true;
-        std::unique_lock<SpinLock> guards[kSkipListMaxHeight];
+
+        auto locked_count = 0;
+        TNode *locked[kSkipListMaxHeight];
+        auto guard = OnScopeExit{[&] {
+          for (auto i = 0; i != locked_count; ++i) {
+            locked[i]->lock.unlock();
+          }
+        }};
+
```

Comment:
Does this give enough of a performance benefit to justify not using the std::unique_lock?

Reply:
Yeah, the expected number of locks is much less than 32. The constructor + destruction of unique_ptr was surprisingly noticable even when it was locking nothing.